pkg/engine/virtualmachinescalesets.go

```diff
@@ -417,7 +417,7 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 	addCustomTagsToVMScaleSets(profile.CustomVMTags, &virtualMachineScaleSet)
 
 	if profile.HasAvailabilityZones() {
-		virtualMachineScaleSet.Zones = &profile.AvailabilityZones
+		virtualMachineScaleSet.Zones = &[]string{fmt.Sprintf("[parameters('%sAvailabilityZones')]", profile.Name)}
```

Comment:
@manishram, I didn't quite give you the right direction here. Because the profile zones param resolves to an array under the hood, which means we're essentially composing an array inside an array. Here's the error:

```
InvalidTemplate - Deployment template parse failed: 'Error reading string. Unexpected token: StartArray. Path '[0]'.'.
```

Instead, we need to manually compose the array one index item at time.

This is a working solution for this entire if block:

```
	if profile.HasAvailabilityZones() {
		zones := []string{}
		for i := range profile.AvailabilityZones {
			zones = append(zones, fmt.Sprintf("[parameters('%sAvailabilityZones')[%d]]", profile.Name, i))
		}
		virtualMachineScaleSet.Zones = &zones
	}
```

Reply:
Okay